rero_ils/modules/acq_orders/extensions.py

```diff
@@ -43,6 +43,20 @@ def pre_load(self, data, loader=None):
         data.pop('total_amount', None)
         data.pop('status', None)
 
+    def pre_delete(self, record, force=False):
+        """Called before a record is deleted.
+
+        :param record: the record metadata.
+        """
+        # For pending orders, we are allowed to delete all of its
+        # line orders without futher checks.
+        # there is no need to check if it is a pending order or not because
+        # the can_delete is already execute in the method self.delete
+        order_record = record.__class__.get_record_by_pid(record.pid)
+        if order_record:
```

Comment:
the `record` method attribute is already the AcqOrder record. No need to reload it (2 DB calls).
Just use 
```
for line in record.get_order_lines():
   line.delete(delindex=True)
```